### PR TITLE
Fixes #23773: Make rudder-package able to uninstall packages

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23773

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/archive.rs [hunk 1]
```diff
@@ -90,7 +88,11 @@ impl Rpkg {
     fn get_txz_dst(&self, txz_name: &str) -> String {
```
Comment (amousset) [inline relay/sources/rudder-package/src/archive.rs:88]: we use the result as a path so returning a PathBuf would make more sense (and make the return less awkward)

File: relay/sources/rudder-package/src/database.rs [hunk 2]
```diff
@@ -38,6 +48,44 @@ impl Database {
...
+            .metadata
+            .run_package_script(PackageScript::Postrm, PackageScriptArg::None)?;
+        // Remove associated package scripts and plugin folder
+        fs::remove_dir_all(
+            PathBuf::from(PACKAGES_FOLDER).join(installed_plugin.metadata.name.clone()),
+        )?;
+        // Update the database
+        let mut updated_db = self.clone();
+        updated_db.plugins.remove(plugin_name);
+        Database::write(PACKAGES_DATABASE_PATH, updated_db)?;
```
Comment (amousset) [inline relay/sources/rudder-package/src/database.rs:86]: shouldn't this update `self` too?

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/database.rs (hunk 2 above)
Comment (Fdall) [reply relay/sources/rudder-package/src/database.rs]: yep

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Pull request merged**